ecide_parts.c

```diff
@@ -121,6 +126,15 @@ static int      ide_probe_adfs_parts(ide_host_t *ih, unsigned int drive,
             ih->card_num, drive,
             bb->log2secsize, bb->secspertrack, bb->heads, bb->disc_size, bb->disc_type,
             &bb->disc_name[0]);
+
+        /* Calculate the offset of the sector following the end of volume and
+        return it in an out-parameter (used for HCCS partitioning scheme where
+        there's no partition table, just a bunch of concatenated ADFS
+        filesystems) */
+        if (out_next_sector_offset != NULL) {
+                *out_next_sector_offset = (sector_offset + bb->disc_size / (2 << (bb->log2secsize-1)));
```

Comment:
The maths here looks kinda sneaky :-) :-)  But also I have no confidence of Norcroft C folding this properly.  Can I suggest simplifying to:
> *out_next_sector_offset = sector_offset + (bb->disc_size >> bb->log2secsize);